src/platform/forms-system/src/js/web-component-patterns/datePatterns.jsx

```diff
@@ -61,6 +64,40 @@ const currentOrPastDateUI = options => {
   };
 };
 
+/**
+ * Web component v3 uiSchema for date range
+ *
+ * ```js
+ * exampleDateRange: dateRangeUI()
+ * exampleDateRange: dateRangeUI({
+ *   fromLabel: 'Start date',
+ *   toLabel: 'End date',
+ * })
+ * ```
+ * @param {string} [fromLabel] - Custom label for the start date field
+ * @param {string} [toLabel] - Custom label for the end date field
+ * @param {string} [errorMessage] - Custom error message for the date range validation
+ * @returns {UISchemaOptions} uiSchema
+ */
+const dateRangeUI = ({
+  fromLabel = 'From date',
+  toLabel = 'To date',
+  errorMessage,
+} = {}) => ({
+  'ui:validations': [
+    {
+      validator: (errors, fieldData, formData, schema, uiSchema) => {
+        validateDateRange(errors, fieldData, formData, schema, uiSchema, {
+          pattern: errorMessage,
+        });
+      },
+    },
+  ],
+  from: currentOrPastDateUI({ title: fromLabel }),
+  to: currentOrPastDateUI({ title: toLabel }),
+  ...(errorMessage && { 'ui:errorMessages': { pattern: errorMessage } }),
```

Comment:
where does this error message show up?

Reply:
This error message or the default error message will appear on the 'to date' field when the date range is not valid.